### PR TITLE
Groupwise start test

### PR DESCRIPTION
**Pull request by czerw**
New test poo#20092: Implement basic groupwise start to catch situation
with possible crash bsc#1045340.

Verification: http://10.100.12.105/tests/1067#step/groupwise/1
Needle: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/424

### CODE REVIEW

File: tests/x11regressions/groupwise/groupwise.pm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+sub run() {
+    select_console('root-console');
+    pkcon_quit;
+
+    # add SLES repository and install dovecot
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/Factory:/Head:/Internal/standard/ groupwise_repo");
+    zypper_call("--gpg-auto-import-keys ref");
+    zypper_call("in novell-groupwise-gwclient", exitcode => [0, 102, 103]);
+    save_screenshot;
```
Comment (mimi1vx) [inline tests/x11regressions/groupwise/groupwise.pm]: please remove this repo after installation

File: tests/x11regressions/groupwise/groupwise.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use strict;
+use base "x11regressiontest";
+use testapi;
+use utils;
+
+sub run() {
+    select_console('root-console');
+    pkcon_quit;
+
+    # add SLES repository and install dovecot
```
Comment (coolo) [inline tests/x11regressions/groupwise/groupwise.pm]: didn't your parents teach you not to lie in comments? tss!

File: tests/x11regressions/groupwise/groupwise.pm (hunk 1 above)
Comment (czerw) [inline tests/x11regressions/groupwise/groupwise.pm]: Repository removal added, but it is not possible to use opensuse for download.

File: tests/x11regressions/groupwise/groupwise.pm (hunk 2 above)
Comment (czerw) [inline tests/x11regressions/groupwise/groupwise.pm]: noticed on lat call, fixed

**Comment by coolo** [comment]
the test looks ok, but your needle is unacceptable

**Comment by coolo** [comment]
please rebase - on top of your other PR

**Comment by czerw** [comment]
Code was rebased.

File: tests/x11regressions/groupwise/groupwise.pm [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+use base "x11regressiontest";
+use testapi;
+use utils;
+
+sub run() {
+    select_console('root-console');
+    pkcon_quit;
+
+    # add repository and install groupwise
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/Factory:/Head:/Internal/standard/ groupwise_repo");
```
Comment (okurz) [inline tests/x11regressions/groupwise/groupwise.pm]: Why should we check packages that are not in SLE?

Comment (mimi1vx) [inline tests/x11regressions/groupwise/groupwise.pm]: because is officially used for internal company communication?

Comment (plusky) [inline tests/x11regressions/groupwise/groupwise.pm]: Well compatibility with 3rd party applications is rather important for SLE. Apart from that running groupwise has uncovered serious kernel regression in past.

Comment (okurz) [inline tests/x11regressions/groupwise/groupwise.pm]: Sure. I am all for testing groupwise but how is "SUSE:Factory:Head:Internal" ensured to be compatible with every version calling this test module, e.g. SLE 12 GA LTSS (or similar in maintenance) up to SLE 15?

Comment (okurz) [inline tests/x11regressions/groupwise/groupwise.pm]: btw, we should follow "factory first" and have a test running for openSUSE first. WDYT?

Comment (czerw) [inline tests/x11regressions/groupwise/groupwise.pm]: Unfortunately, groupwise client has SUSE-NonFree license, so there is no way to get it to public repository for openSUSE(or not?).  Also, it is just repackaged another rpm file with jar files, without any source compilation.  SUSE:Factory:Head:Internal repository (maintained by brave volunteers) is the only repository recommended by the company to employees on wiki https://wiki.microfocus.net/index.php?title=GroupWise_Linux .
It means that we are in "best effort" situation and we would like to have at least basic test coverage.

Comment (okurz) [inline tests/x11regressions/groupwise/groupwise.pm]: Well, there is https://en.opensuse.org/Package_repositories#Non-OSS

Also, interesting enough it seems groupwise has been part of SLE-11 and then dropped maybe?

So I understand we can't use this test for openSUSE, fine. How about testing it on different versions of SLE though? You only tested SLE12SP2 so far it seems. Please check at least SLE12SP3 then

Comment (czerw) [inline tests/x11regressions/groupwise/groupwise.pm]: SP3: http://10.100.12.105/tests/1100#step/groupwise/19
Unfortunately, I can't comment much history of groupwise (i'm not using it and definitely never will).
Maybe we can open discussion with guys who do update on the package from time to time.

**Comment by okurz** [comment]
+1

**Comment by mimi1vx** [comment]
@czerw rebase please

**Comment by czerw** [comment]
@mimi1vx rebased

**Comment by plusky** [comment]
So, its four days since review and still not merged ...

**Comment by okurz** [comment]
yes, and probably it helps if we try to understand the reasons. Af first, four days is not particularly long for the current project. Sure, would be great to have faster feedback. IMHO the best way to achieve that is to encourage more people to do the peer-review. Second, this is a "new test", not a fix to existing tests which IMHO should always be favored. Also, based on the discussion within this PR I guess not enough people feel confident to merge it. Also myself I wanted to have more "buy-in" from others, hence only "+1" by me, not immediate approval and/or merge.

**Comment by plusky** [comment]
@okurz - I am not under impression that "buy in" or convincing people about usefulness of new tests is necessary - and I do not see potential issues caused by this new tests nor were any raised in discussion. Reasoning why QAM wants this test is present, and I do not see it being challanged - and four days waiting makes planning of resources challanging - can asignee of task start working on something else or does he need to wait for more feedback

**Comment by okurz** [comment]
@plusky I agree with your concerns. I did not see a clear "LGTM" by anyone or an obvious approval. Hence I did not merge it. There are so many that could have merged it and I would not have objected if they did. You did not give a LGTM then but I read your comments as equivalent so merging :-)

**Comment by coolo** [comment]
Note that the argumentation seems a bit personal targeted vs @okurz but there are ~15 people able to merge tests (new and fixes) and okurz was the only one reviewing it (and felt unsure about it, so didn't merge it right away). So you should look at why the other 14 didn't act - 3 of them sit in the same office room as you from what I remember.

**Pull request merged**